CONTRIBUTING.md

```diff
@@ -48,20 +79,24 @@ The following is a short list of commands that can be run in the root of the pro
 
 We provide a git pre-commit hook to format and check the code, to install it run `make link-git-hooks`.
 
-### Generating Mocks
+#### Generating Mocks
 
 We use [mockgen](https://github.com/golang/mock) to handle mocking in our unit tests.
-If you need a new mock please update or add the `//go:generate` instruction to the appropriate file. 
+If you need a new mock please update or add the `//go:generate` instruction to the appropriate file.
 
 ### Adding a New Command
 
 `mongocli` uses [Cobra](https://github.com/spf13/cobra) as a framework for defining commands,
 in addition to this we have defined a basic structure that should be followed.
-For a `mongocli scope newCommand` command a file `internal/cli/scope_new_command.go` should implement: 
+For a `mongocli scope newCommand` command a file `internal/cli/scope/new_command.go` should implement: 
 - A `ScopeNewCommandOpts` struct which handles the different options for the command.
 - At least a `func (opts *ScopeNewCommandOpts) Run() error` function with the main command logic.
 - A `func ScopeNewCommandBuilder() *cobra.Command` function to put together the expected cobra definition along with the `ScopeNewCommandOpts` logic.
 
+Commands try to follow a [RESTful](https://en.wikipedia.org/wiki/Representational_state_transfer) approach to match the APIs, 
+this means arguments to the command try to match path and query params where the last path param will be a required argument of the command and teh rest will be handled via flag options.
+For commands that create or modify complex data structure the use of configuration files is preferred over flag options 
+
```

Comment:
A slight rephrasing and punctuation suggestion:

Commands follow a [RESTful](https://en.wikipedia.org/wiki/Representational_state_transfer) approach to match the APIs, whenever possible. For that reason, command arguments tend to match the path and query params of the APIs, with the last param being a required argument and the rest handled via flag options. For commands that create or modify complex data structures, the use of configuration files is preferred over flag options.

Reply:
thanks, I really wanted comments on this paragraph and this is way better